context/app/models/user.rb

```diff
@@ -24,4 +25,8 @@ def before_save_hash_password
     end
   end
 
+  def after_create_ensure_name
+    update_attribute :name, "user#{id}" if name.blank?
```

Comment:
The idea is to name user by concatenating 'user' and user.id

There is no way to do it without extra query :(